DNN Platform/Library/Entities/Tabs/TabController.cs

```diff
@@ -1469,6 +1469,33 @@ public TabCollection GetTabsByPortal(int portalId)
                                                             });
         }
 
+        /// <summary>
+        /// Get the actual visible tabs for a given portal id. 
+        /// System Tabs and Admin Tabs are excluded from the result set.
+        /// </summary>
+        /// <param name="portalId"></param>
+        /// <param name="adminTabId"></param>
+        /// <returns></returns>
+        public TabCollection GetTabsByPortal(int portalId, int? adminTabId)
+        {
+            var tabs = GetTabsByPortal(portalId);

```

Comment:
I think that @mitchelsellers' idea was to have two separate keys `{ portalId }` and `{ portalId, adminTabId }` rather than retrieving one and filtering in memory. However, for a functionality probably scarcely used I think we can live with that

Reply:
I agree @tpluscode I would have expected a separate key and object in cache, just not sure of the usage.  Given the strong usage of cache in other locations though, I would expect this to cache results as well.

Reply:
Well, we cannot avoid filtering in memory without more changes to CBO. I think the current solution is good enough where we reuse the existing cache entry